### PR TITLE
[DPTP-2635]cluster-display: add endpont for clusters

### PR DESCRIPTION
**Pull request by hongkailiu**
```console
$ curl -s "localhost:8090/api/v1/clusters" | jq
{
  "data": [
    {
      "cloud": "AWS",
      "cluster": "app.ci",
      "consoleHost": "console-openshift-console.apps.ci.l2s4.p1.openshiftapps.com",
      "product": "OSD",
      "registryHost": "registry.ci.openshift.org",
      "version": "4.8.3"
    },
    {
      "cloud": "AWS",
      "cluster": "arm01",
      "consoleHost": "console.arm-build01.arm-build.devcluster.openshift.com",
      "product": "OCP",
      "registryHost": "registry.arm-build01.arm-build.devcluster.openshift.com",
      "version": "4.9.0-0.nightly-arm64-2021-07-05-165629"
    },
    {
      "cloud": "AWS",
      "cluster": "build01",
      "consoleHost": "console.build01.ci.openshift.org",
      "product": "OCP",
      "registryHost": "registry.build01.ci.openshift.org",
      "version": "4.9.11"
    },
    {
      "cloud": "GCP",
      "cluster": "build02",
      "consoleHost": "console.build02.ci.openshift.org",
      "product": "OCP",
      "registryHost": "registry.build02.ci.openshift.org",
      "version": "4.9.10"
    },
    {
      "cloud": "AWS",
      "cluster": "build03",
      "consoleHost": "console-openshift-console.apps.build03.ky4t.p1.openshiftapps.com",
      "product": "OSD",
      "registryHost": "registry.build03.ci.openshift.org",
      "version": "4.8.13"
    },
    {
      "cloud": "GCP",
      "cluster": "build04",
      "consoleHost": "console-openshift-console.apps.build04.34d2.p2.openshiftapps.com",
      "product": "OSD",
      "registryHost": "registry.build04.ci.openshift.org",
      "version": "4.9.6"
    },
    {
      "cloud": "AWS",
      "cluster": "hive",
      "consoleHost": "console-openshift-console.apps.hive.9xw5.p1.openshiftapps.com",
      "product": "OSD",
      "registryHost": "default-route-openshift-image-registry.apps.hive.9xw5.p1.openshiftapps.com",
      "version": "4.9.11"
    },
    {
      "cloud": "AWS",
      "cluster": "vsphere",
      "consoleHost": "console-openshift-console.apps.build01-us-west-2.vmc.ci.openshift.org",
      "product": "OCP",
      "registryHost": "registry.apps.build01-us-west-2.vmc.ci.openshift.org",
      "version": "4.8.20"
    }
  ]
}
```

### CODE REVIEW

**Comment by hongkailiu** [comment]
launch 4.9.0-0.okd-2021-11-28-035710

**Comment by hongkailiu** [comment]
/hold

Need to go together with https://github.com/openshift/release/pull/24421

File: cmd/cluster-display/main.go [hunk 1]
```diff
@@ -169,18 +279,44 @@ func main() {
...
-	if len(kubeConfigs) != 1 || !ok {
-		logrus.WithError(err).Fatalf("found %d contexts in Hive kube config and it must be %s", len(kubeConfigs), string(api.HiveCluster))
+
+	inClusterConfig, hasInClusterConfig := kubeConfigs[kube.InClusterContext]
+	delete(kubeConfigs, kube.InClusterContext)
+	delete(kubeConfigs, kube.DefaultClusterAlias)
+
+	if _, hasAppCi := kubeConfigs[appCIContextName]; !hasAppCi {
+		if !hasInClusterConfig {
+			logrus.WithError(err).Fatalf("had no context for '%s' and loading InClusterConfig failed", appCIContextName)
```
Comment (smg247) [inline cmd/cluster-display/main.go:289]: can we replace all these logging calls to a logger with fields?

File: cmd/cluster-display/main.go [hunk 2]
```diff
@@ -60,6 +67,12 @@ func addSchemes() error {
 	if err := hivev1.AddToScheme(scheme.Scheme); err != nil {
 		return fmt.Errorf("failed to add hivev1 to scheme: %w", err)
 	}
+	if err := routev1.AddToScheme(scheme.Scheme); err != nil {
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/cluster-display/main.go]: Updated.

File: cmd/cluster-display/main.go [hunk 3]
```diff
@@ -103,7 +116,72 @@ func getClusterPoolPage(ctx context.Context, hiveClient ctrlruntimeclient.Client
...
+		if !kerrors.IsNotFound(err) {
+			return "", fmt.Errorf("failed to get Service %s in namespace %s: %w", name, ns, err)
+		}
+	} else {
+		return "OSD", nil
+	}
+	if strings.Contains(version, "okd") {
+		return strings.ToUpper(string(api.ReleaseProductOKD)), nil
+	}
+	return strings.ToUpper(string(api.ReleaseProductOCP)), nil
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/cluster-display/main.go]: Updated.

File: cmd/cluster-display/main.go (hunk 1 above)
Comment (hongkailiu) [reply cmd/cluster-display/main.go]: Updated.

File: pkg/api/domain.go [hunk 4]
```diff
@@ -61,3 +67,60 @@ func RegistryDomainForClusterName(clusterName string) (string, error) {
...
 	return "", fmt.Errorf("failed to get the domain for cluster %s", clusterName)
 }
+
+// ResolveConsoleHost resolves the console host
+func ResolveConsoleHost(ctx context.Context, client ctrlruntimeclient.Client) (string, error) {
+	routes := &routev1.RouteList{}
+	namespace := "openshift-console"
+	if err := client.List(ctx, routes, ctrlruntimeclient.InNamespace(namespace)); err != nil {
+		return "", fmt.Errorf("failed to list routes in namespace %s: %w", namespace, err)
+	}
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply pkg/api/domain.go:77]: I feel the extraction does not simplify the code much and it is easier to read this way.
I would like to keep this as it is if it is OK with you.

[the remark this answers is not in the document]
Comment (smg247) [reply pkg/api/domain.go:77]: Thats fine

**Review by smg247 — APPROVED** [review]

**Pull request merged**